pkg/clustermarshaller/clustermarshaller.go

```diff
@@ -28,6 +28,10 @@ func MarshalClusterSpec(clusterSpec *cluster.Spec, datacenterConfig providers.Da
 		marshallables = append(marshallables, clusterSpec.GitOpsConfig.ConvertConfigToConfigGenerateStruct())
 	}
 
+	if clusterSpec.FluxConfig != nil {
```

Comment:
Should we have an if/else to only marshal either `GitOpsConfig` or `FluxConfig`?

Reply:
yeah, I added a check that throws an error if both are present, and filed at ticket to add a higher-level check for mutual exclusivity. Additionally, added a test which checks that if both are present we throw an error.